### ISSUE TITLE
`shuffle()` does not work with structure field

### ISSUE THREAD

**Issue by texnixe** [getkirby/kirby#288]
Don't know if this is a bug or simply not implemented, but I was expecting `shuffle(`) to work in conjunction with the new `toStructure()` method on structure fields:

```php
$items = $page->items()->toStructure()->shuffle();
```

like with page collections. But it does not work.

**Comment by bastianallgeier** [comment]
I just checked and it works here.

**Comment by bastianallgeier** [comment]
I'm closing this as it seems to be fixed.

**Comment by texnixe** [comment]
I'm reopening this, because I don't think this works, at least I could never get it to work.

I have a simple structure field which currently only consists of one field:

```
fields:
  slogans:
    label: Slogans
    type: structure
     fields:
      slogan_title:
        label: Slogan
        type: text
```

Ok, now I can easily get the first slogan like in any collection:

```
$slogan = $page->slogans()->toStructure()->first();
echo $slogan->slogan_title();
```

But if I try to shuffle my little Slogan collection first, it does not work:

```
$slogan = $page->slogans()->toStructure()->shuffle()->first();
echo $slogan->slogan_title();
```
results in an error message: Fatal error: Call to a member function slogan_title() on integer in ...

But even the `shuffle()`method without `first()` doesn't seem to do what it should, if it were a proper collection.
Kirby 2.2.3 develop version.

**Comment by AhoyLemon** [comment]
Currently having this problem with a site as well:
`$randomLucky = $page->lucky_people()->toStructure()->shuffle()->first();` doesn't work, but `$randomLucky = $page->lucky_people()->toStructure()->first();` does.

**Comment by texnixe** [comment]
I'd like to push this up again on the agenda, there are quite a few people on the forum recently coming across this issue. All we can tell them is to use `yaml()` and then shuffle the resulting array.

**Comment by lukasbestle** [comment]
I'm working on this.

**Comment by lukasbestle** [comment]
I fixed this in the Toolkit, please see https://github.com/getkirby/toolkit/pull/182.
It would be very helpful if you guys could help testing if it works now.

**Comment by lukasbestle** [comment]
This is now included on the develop branch of the Toolkit and will be fixed in Kirby 2.3.1.

**Comment by Thiousi** [comment]
Brilliant !
